lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_producer.rb

```diff
@@ -101,6 +102,15 @@ def join_enterprise_fee
           end
         end
 
+        def enterprise_fees_related_to_incoming_exchanges_ids(order)
+          order
+            .order_cycle
+            .exchanges
+            .incoming
+            .flat_map(&:enterprise_fee_ids)
+            .uniq
+        end
```

Comment:
It would be great to merge this query into the above one to optimise it; but I think it's too complex. 
This does the job, well done 👍